src/Nethermind/Nethermind.Blockchain/Comparers/GasPriceTxComparerHelper.cs

```diff
@@ -28,8 +28,12 @@ public static int Compare(Transaction? x, Transaction? y, UInt256 baseFee, bool
             if (ReferenceEquals(null, y)) return 1;
             if (ReferenceEquals(null, x)) return -1;
             
-            // then by gas price descending
-            
+            // if gas bottleneck was calculated, it's highest priority for sorting
+            if (x.GasBottleneck != UInt256.MaxValue || y.GasBottleneck != UInt256.MaxValue)
```

Comment:
I don't like magic value here

Reply:
I changed initial value of GasBottleneck to 0 and now I'm comparing here to 0 instead to MaxValue. Still I want to use GasBottleneck comparator only when GasBottleneck was already calculated so need to check it somehow